Jellyfin.Server/Migrations/MigrationRunner.cs

```diff
@@ -81,5 +85,42 @@ public static void Run(CoreAppHost host, ILoggerFactory loggerFactory)
                 logger.LogDebug("Migration '{Name}' marked as applied in configuration.", migrationRoutine.Name);
             }
         }
+
+        /// <summary>
+        /// Runs the network setting migration, which has to be done before settings.xml is loaded.
+        /// </summary>
+        /// <param name="appPaths">The <see cref="IServerApplicationPaths"/>.</param>
+        /// <param name="logger">The <see cref="ILogger"/>.</param>
+        public static void RunNetworkSettingMigration(IServerApplicationPaths appPaths, ILogger logger)
+        {
+            var destFile = Path.Combine(appPaths.ConfigurationDirectoryPath, "network.xml");
+            if (!File.Exists(destFile))
+            {
+                try
+                {
+                    var xmlSerializer = new MyXmlSerializer();
+                    var source = xmlSerializer.DeserializeFromFile(typeof(Legacy.ServerConfiguration), appPaths.SystemConfigurationFilePath);
+
+                    var target = new NetworkConfiguration();
```

Comment:
This should be its own "legacy" class as well.

Reply:
It is, (sort of)- it's Legacy.ServerConfiguration.

Legacy.ServerConfiguration contains the properties are mutual to the 10.7 versions of system.xml and network.xml.

The settings subsystem loads the current xml, discards and changes and re-saves a clean on if there is anything "extra"
Storing into the 10.X version of network.xml here, just saves a resave by the settings subsystem.

Reply:
The point I'm trying to make is that migrations should be version agnostic. To do that we need to migrate the settings using only what we know in 10.7. The changes I'm requesting are sort of paving the way for future config migrations (we will need a dedicated migration runner when that time comes though).

Reply:
This should then work, as only the properties present in _Legacy.ServerConfiguration_ should get migrated across into the current structure. 

The only additional safeguard I can possibily think of, is to wrap the setValue (line 112) in its own try/catch block just in case the _datatype_ of a property changes in the future.

Reply:
`current structure` is ephemeral though. If this migration is executed in 10.10 it might have different options that were migrated between other releases. Information will be lost in this case.

Reply:
so you suggesting this should be static, and updated every release if the structure changes?

Reply:
No, this is a one-off migration. Any other changes will need a _new_ migration, which is why I also said we'd need a `configuration migration runner` some time in the future (overkill for now, _I think_).

Reply:
Sorry, but I think I'm still missing something. This is how I see the process.

ServerConfiguration is ephemeral, so to overcome this Legacy.ServerConfiguration has been created - which is basically a fixed point in time 10.7.0 snapshot of NetworkConfiguration.

NetworkConfiguration is ephemeral. So only the values that appear both in Legacy.ServerConfiguration and NetworkConfiguration are copied across. 

If NetworkConfiguration changes in the future then under this PR the properties won't copy.  A new migration PR would be required.

If Legacy.ServerConfiguration were copied into Legacy.NetworkConfiguration, and NetworkConfiguration changes - the issues will still remain, with invalid settings in Legacy.NetworkConfiguration begin dropped.

Reply:
> If Legacy.ServerConfiguration were copied into Legacy.NetworkConfiguration, and NetworkConfiguration changes - the issues will still remain, with invalid settings in Legacy.NetworkConfiguration begin dropped.

In this scenario, _other_ migrations would handle it. After all the migrations have run, the resulting `NetWorkConfiguration` would be whole and valid. This migration is just a small cog.